main/src/ca/uwaterloo/flix/language/phase/jvm/GenMainClass.scala

```diff
@@ -19,20 +19,66 @@ package ca.uwaterloo.flix.language.phase.jvm
 import ca.uwaterloo.flix.api.Flix
 import ca.uwaterloo.flix.language.ast.FinalAst.{Def, Root}
 import ca.uwaterloo.flix.language.ast.Symbol
+import org.objectweb.asm.ClassWriter
+import org.objectweb.asm.Opcodes._
 
 /**
   * Generates bytecode for the main class.
   */
 object GenMainClass {
 
+
   /**
     * Returns the main class.
     */
   def gen()(implicit root: Root, flix: Flix): Map[JvmName, JvmClass] = getMain(root) match {
     case None => Map.empty
     case Some(defn) =>
       // TODO: Ramin: Emit a main class with the appropriate method that calls the main shim.
-      Map.empty
+      val jvmType = JvmOps.getMainClassType()
+      val jvmName = jvmType.name
+      val targs = List[JvmType](JvmType.Array(JvmType.String))
+      val bytecode = genByteCode(jvmType, targs)
+      Map(jvmName -> JvmClass(jvmName, bytecode))
+  }
+
+
+
+  def genByteCode(jvmType: JvmType.Reference, targs: List[JvmType])(implicit root: Root, flix: Flix): Array[Byte] = {
+    // class writer
+    val visitor = AsmOps.mkClassWriter()
+
+    // internal name of super
+    val superClass = JvmName.Object.toInternalName
+
+    // Initialize the visitor to create a class.
+    visitor.visit(AsmOps.JavaVersion, ACC_PUBLIC + ACC_FINAL, jvmType.name.toInternalName, null, superClass, null)
+
+    // Source of the class
+    visitor.visitSource(jvmType.name.toInternalName, null)
+
+    // Emit the code for the main method
+    compileMainMethod(visitor, jvmType, targs)
+
+    visitor.visitEnd()
+    visitor.toByteArray
+  }
+
+  def compileMainMethod(visitor: ClassWriter, jvmType: JvmType.Reference, targs: List[JvmType])(implicit root: Root, flix: Flix): Unit = {
+
+    val main = visitor.visitMethod(ACC_PUBLIC + ACC_STATIC, "main", AsmOps.getMethodDescriptor(targs, JvmType.Void), null, null)
+
+    main.visitCode()
+    main.visitTypeInsn(NEW, JvmType.Context.name.toInternalName)
+    main.visitInsn(DUP)
+
+    main.visitMethodInsn(INVOKESPECIAL, JvmName.Context.toInternalName,
+      "<init>", AsmOps.getMethodDescriptor(Nil, JvmType.Void), false)
+
+
+    main.visitMethodInsn(INVOKEVIRTUAL, JvmName.Context.toInternalName, "m_main", AsmOps.getMethodDescriptor(Nil, JvmType.Void), false)
```

Comment:
Here you need to load the appropriate namespace object and then get the m_main from it